lib/active_publisher/message.rb

```diff
@@ -1,3 +1,20 @@
+require "json"
 module ActivePublisher
-  class Message < Struct.new(:route, :payload, :exchange_name, :options); end
+  class Message < Struct.new(:route, :payload, :exchange_name, :options)
+    class << self
+      def from_json(payload)
+        parsed = JSON.load(payload)
+        self.new(
+          parsed["route"],
+          parsed["payload"],
+          parsed["exchange_name"],
+          parsed["options"],
+        )
+      end
+    end
+
+    def to_json
+      self.to_h.to_json
```

Comment:
We will probably want to wrap `"payload"` in base64 encoding to avoid parsing random bytes as a string in json (won't work).